### DISCUSSION TITLE
small improvement of tree-sitter parser installation

### DISCUSSION THREAD

**Discussion by mlyxshi** [malob/nix-config#15] [Ideas]
https://github.com/malob/nixpkgs/blob/1540c7b83ae86da9c549593a21744a60e2e41252/home/neovim.nix#L91

After GCC or it dependencies upgrade, tree-sitter will not work due to library path change.

Another way(give up hm neovim module)
```
let
  TREESITTER = (pkgs.vimPlugins.nvim-treesitter.withPlugins (
    plugins: with plugins; [
      tree-sitter-bash
      tree-sitter-json
      tree-sitter-typescript
      tree-sitter-c
      tree-sitter-cpp
      tree-sitter-nix
      tree-sitter-lua
      tree-sitter-rust
      tree-sitter-python
      tree-sitter-markdown
    ]
  ));

in
{

  xdg.configFile = {
    "nvim/init.lua".text = ''
      -- Begin Stage: Cache and Accelerate
      require("plugin-config.impatient")
      
      -- Add Treesitter Path
      vim.opt.runtimepath = vim.opt.runtimepath + "${TREESITTER}"
      
      -- Invoke Real Start
      require("start")
    '';
  } ;

}
```

Learn a lot from your config, thanks!

**Comment by malob** [comment]
Thanks! 

I've updated my config to use `nvim-treesitter.withPlugins` and it seems to be working (https://github.com/malob/nixpkgs/commit/a0cfabd0e416a7b48acf5422b49ad542e5ac7089)

Separately, I noticed in the readme for your config you say:
> Darwin is my main OS. 
> Therefore, I will not use {xdg.configFile."file".source} to manage config  <-- Everytime I make a minor change, I have to rebuild my MacOS
> Instead, I decide to use the conventional and undeterministic way in MacOS <-- Simple symbolic link

I also find this annoying for configuration files I change frequently. Do you know about `config.lib.file.mkOutOfStoreSymlink`? It's a great hack for exactly this issue. For example: https://github.com/malob/nixpkgs/blob/b7b2a825a1f4e3f7ec8c7614bd32f3a5bfa2ddd1/home/neovim.nix#L36-L37

**Reply by mlyxshi** [reply]
`mkOutOfStoreSymlink ` works very well. Thanks
